### PR TITLE
Updated so that the dhcp lease poller parsed data does not include expired leases

### PR DESCRIPTION
**Pull request by tldavies**

### CODE REVIEW

**Comment by tldavies** [comment]
@DavidjohnBlodgett  @RackHD/corecommitters

File: lib/jobs/isc-dhcp-lease-poller.js [hunk 1]
```diff
@@ -144,6 +144,10 @@ function iscDhcpLeasePollerFactory(
         */
         var split = data.split('\n');
         var lease;
+        var todaysDate = new Date();
```
Comment (benbp) [inline lib/jobs/isc-dhcp-lease-poller.js]: Can we call this currentTime or something? `todaysDate` just makes me think of a date at the day level, not one that's precise to the millisecond.

File: lib/jobs/isc-dhcp-lease-poller.js [hunk 2]
```diff
@@ -144,6 +144,10 @@ function iscDhcpLeasePollerFactory(
...
+        var notExpired = false;
+
+       //look at cat /var/lib/dhcp/dhcpd.leases to find leases
```
Comment (benbp) [inline lib/jobs/isc-dhcp-lease-poller.js]: I would get rid of the `cat` in this sentence.

File: lib/jobs/isc-dhcp-lease-poller.js [hunk 3]
```diff
@@ -157,6 +161,18 @@ function iscDhcpLeasePollerFactory(
                 return ip;
             }
 
+            if(line.startsWith('ends')) {
+                // slice off the to just get the date and time
+                var endTime = line.slice(7, -1);
+
+                var leaseDate = new Date(endTime);
```
Comment (benbp) [inline lib/jobs/isc-dhcp-lease-poller.js]: Another naming comment: perhaps `expirationDate`?

File: lib/jobs/isc-dhcp-lease-poller.js [hunk 4]
```diff
@@ -144,6 +144,10 @@ function iscDhcpLeasePollerFactory(
         */
         var split = data.split('\n');
         var lease;
+        var todaysDate = new Date();
+        var notExpired = false;
```
Comment (benbp) [inline lib/jobs/isc-dhcp-lease-poller.js]: This double negative is also kind of confusing, can we just do: `var expired = true`?

File: lib/jobs/isc-dhcp-lease-poller.js [hunk 5]
```diff
@@ -157,6 +161,18 @@ function iscDhcpLeasePollerFactory(
...
 
+            if(line.startsWith('ends')) {
+                // slice off the to just get the date and time
+                var endTime = line.slice(7, -1);
+
+                var leaseDate = new Date(endTime);
+
+                //Check to see if it is not expired.
+                if(leaseDate > todaysDate) {
+                    notExpired = true;
```
Comment (benbp) [inline lib/jobs/isc-dhcp-lease-poller.js]: Actually, is this variable even necessary? Couldn't you just `return ip` here if the conditional is `leaseDate < todaysDate`?

File: spec/lib/jobs/isc-dhcp-poller-job-spec.js [hunk 6]
```diff
@@ -75,12 +126,35 @@ describe('ISC DHCP Poller Job', function () {
...
         });
+
+        it('should not parse an expired lease', function() {
+            var job = new this.Jobclass({}, {}, uuid.v4());
+            var parsed = job.parseLeaseData(expiredLeaseData.toString());
+
+            expect(parsed).to.be.undefined;
+        });
+
+        it('should only parse none expired leases', function() {
```
Comment (benbp) [inline spec/lib/jobs/isc-dhcp-poller-job-spec.js]: "none expired leases"?

File: lib/jobs/isc-dhcp-lease-poller.js (hunk 1 above)
Comment (tldavies) [inline lib/jobs/isc-dhcp-lease-poller.js]: Sure

File: lib/jobs/isc-dhcp-lease-poller.js (hunk 3 above)
Comment (tldavies) [inline lib/jobs/isc-dhcp-lease-poller.js]: have about expiredtime as you didnt seem to like date before?

File: lib/jobs/isc-dhcp-lease-poller.js (hunk 5 above)
Comment (tldavies) [inline lib/jobs/isc-dhcp-lease-poller.js]: if you return the ip here it will still go to the next line. Which will add the lease.

**Comment by benbp** [comment]
:+1:

**Comment by DavidjohnBlodgett** [comment]
:+1:

**Pull request merged**